### PR TITLE
Search Service to support search for folders available for alerting

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
This PR introduced support for a new query type in search service `dash-folder-alerting`. The default behavior of this type is the same as `dash-folder`. However, in the case of fine-grained access, this lets the user search for folders that are available for the customer to read\write alerts. 

**Which issue(s) this PR fixes**:
Related to: https://github.com/grafana/grafana/pull/45749

**Special notes for your reviewer**:
- [x] Tests

### CODE REVIEW

File: pkg/services/sqlstore/permissions/dashboard.go [hunk 1]
```diff
@@ -77,37 +78,62 @@ func (d DashboardPermissionFilter) Where() (string, []interface{}) {
...
+// NewAccessControlDashboardPermissionFilter creates a new AccessControlDashboardPermissionFilter that is configured with specific actions calculated based on the models.PermissionType and query type
+func NewAccessControlDashboardPermissionFilter(user *models.SignedInUser, permissionLevel models.PermissionType, queryType string) AccessControlDashboardPermissionFilter {
+	needEdit := permissionLevel > models.PERMISSION_VIEW
+	var folderActions []string
+	var dashboardActions []string
+	if queryType == searchstore.TypeAlertFolder {
+		folderActions = append(folderActions, accesscontrol.ActionAlertingRuleRead)
+		if needEdit {
+			folderActions = append(folderActions, accesscontrol.ActionAlertingRuleUpdate)
+		}
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/permissions/dashboard.go:95]: As discussed in the meeting yesterday, I think we still need to add `dashboards.ActionFoldersRead` for this query. @kalleep to confirm :)

File: pkg/services/ngalert/api/authorization.go [hunk 2]
```diff
@@ -75,43 +49,43 @@ var (
...
-					Action: ActionAlertingRuleUpdate,
-					Scope:  ScopeDatasourcesAll,
+					Action: ac.ActionAlertingRuleUpdate,
+					Scope:  datasources.ScopeDatasourcesAll,
 				},
 				{
-					Action: ActionAlertingRuleDelete,
-					Scope:  ScopeDatasourcesAll,
+					Action: ac.ActionAlertingRuleDelete,
+					Scope:  datasources.ScopeDatasourcesAll,
```
Comment (IevaVasiljeva) [inline pkg/services/ngalert/api/authorization.go:73]: question (not related to this PR): I didn't realise that we also had introduced datasource scopes to alerting. Is the plan to use folder and datasource scopes interchangeably?

**Review by IevaVasiljeva — COMMENTED** [review]
Overall looks good, apart from missing the folder read permission.

Nice work! 💪

File: pkg/services/ngalert/api/authorization.go (hunk 2 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/api/authorization.go:73]: Grafana supports two alerting APIs: Grafana Alerting - in-process alerting engine, external providers (Prometheus, Loki, Cortex, Alermanager). API that server Grafana Alerting is protected with folder scope. External Providers - datasource scope.

Comment (IevaVasiljeva) [reply pkg/services/ngalert/api/authorization.go]: I see. I think we currently don't have any other cases where we store two different scopes for the same resource in the DB yet, but hopefully it won't cause issues 🤞

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**